### PR TITLE
Find legal adviser map + results

### PR DESCRIPTION
**Pull request by tyom**

### CODE REVIEW

**Comment by tyom** [comment]
retest this please

**Comment by axemonkey** [comment]
Can't speak for the Python, but the FE LGTM.

File: cla_public/apps/base/filters.py [hunk 1]
```diff
@@ -14,3 +16,17 @@ def datetime(dt, format='medium', locale='en_GB'):
     elif format == 'short':
         format = "dd/MM/y, h:mma"
     return format_datetime(dt, format, locale=locale)
+
+
+@base.app_template_filter()
+def humanize_url(value, trim_url_limit=None, nofollow=None, target=None):
+    pattern = re.compile('(^https?:\/\/)|(\/$)')
```
Comment (andyhd) [inline cla_public/apps/base/filters.py]: if you use a regex string `r'(^https?://)|(/$)'`, you don't need to escape things

Comment (tyom) [inline cla_public/apps/base/filters.py]: [Updated](https://github.com/ministryofjustice/cla_public/commit/78cdcac6a6e5a71145692c8ee6651e75e869812f).

**Comment by tyom** [comment]
retest this please

**Comment by tyom** [comment]
retest this please

**Comment by tyom** [comment]
test this please

**Comment by tyom** [comment]
test this please

**Comment by tyom** [comment]
test this please

File: cla_public/apps/base/filters.py [hunk 2]
```diff
@@ -14,3 +16,20 @@ def datetime(dt, format='medium', locale='en_GB'):
     elif format == 'short':
         format = "dd/MM/y, h:mma"
     return format_datetime(dt, format, locale=locale)
+
+
+@base.app_template_filter()
+def url_to_human(value, trim_url_limit=None, nofollow=None, target=None):
+    return re.sub(r'(^https?://)|(/$)', '', value)
+
```
Comment (andyhd) [inline cla_public/apps/base/filters.py]: need another blank line here for PEP8

File: cla_public/apps/base/filters.py [hunk 3]
```diff
@@ -14,3 +16,20 @@ def datetime(dt, format='medium', locale='en_GB'):
...
+@base.app_template_filter()
+def human_to_url(value, trim_url_limit=None, nofollow=None, target=None):
+    return re.sub(r'^((?!https?://).*)', r'http://\1', value)
```
Comment (andyhd) [inline cla_public/apps/base/filters.py]: These functions both have unused parameters `trim_url_limit`, `nofollow` and `target`

File: cla_public/libs/laalaa.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import requests
+import json
+from flask import current_app
+
```
Comment (andyhd) [inline cla_public/libs/laalaa.py]: 2 blank lines needed her per PEP8

File: cla_public/libs/laalaa.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+class LaaLaaError(Exception):
+    pass
+
```
Comment (andyhd) [inline cla_public/libs/laalaa.py]: And here

File: cla_public/libs/laalaa.py (hunk 4 above)
Comment (tyom) [inline cla_public/libs/laalaa.py]: Done.

**Comment by andyhd** [comment]
:+1:

**Comment by tyom** [comment]
test this please

**Comment by andyhd** [comment]
retest this please

**Comment by andyhd** [comment]
retest this please

**Comment by andyhd** [comment]
retest this please

**Comment by andyhd** [comment]
retest this please

**Comment by andyhd** [comment]
retest this please

**Comment by andyhd** [comment]
retest this please

**Pull request merged**